SwiftUIHomeworks/SwiftUIHomeworks/Screens/GreetingView.swift

```diff
@@ -0,0 +1,192 @@
+                Spacer()
+                logoImageView
+                Spacer()
+                getStartedButtonView
+                navigationLinkProductView
+                havingAccountView
+                signInButtonView
+                navigationLinkAuthenticationView
+                Spacer()
+            }
+            .frame(width: UIScreen.main.bounds.width)
+            .ignoresSafeArea()
+            .background(backgroundLinearGradientView)
+        }
+    }
+
+    private var navigationLinkProductView: some View {
+        NavigationLink(
+            destination: ProductView(),
+            tag: Constants.productViewTagString,
+            selection: $greetingViewModel.tagSelection
+        ) {
+            EmptyView()
+        }
+    }
+
+    private var navigationLinkAuthenticationView: some View {
+        NavigationLink(
+            destination: AuthenticationView(),
+            tag: Constants.authenticationViewTagString,
+            selection: $greetingViewModel.tagSelection
+        ) {
+            EmptyView()
+        }
+    }
+
+    private var titleView: some View {
+        Text(Constants.furnitureTitle.uppercased())
+            .multilineTextAlignment(.center)
+            .font(.system(size: Constants.sizeTitleFontValue, weight: .bold, design: .default))
+            .frame(width: Constants.widthTitleValue)
+            .foregroundColor(.white)
+            .ignoresSafeArea(.all, edges: .top)
+            .offset(y: Constants.offsetYTitleValue)
+            .padding(.top, Constants.offsetYTitleValue)
+    }
+
+    private var logoImageView: some View {
+        AsyncImage(url: greetingViewModel.getUrl()) { phase in
+            switch phase {
+            case .empty:
+                ProgressView()
+            case let .success(image):
+                image
+                    .resizable()
+                    .aspectRatio(contentMode: .fit)
+                    .offset(y: Constants.offsetYLogoImageValue)
+            case .failure:
+                Image(systemName: Constants.failureImageName)
+            @unknown default:
+                EmptyView()
+            }
+        }
+        .frame(width: Constants.widthLogoImageValue, height: Constants.heightLogoImageValue)
+    }
+
+    private var getStartedButtonView: some View {
+        Button {
+            greetingViewModel.tagSelection = Constants.productViewTagString
+        } label: {
+            Text(Constants.getStartedButtonTitle.uppercased())
+                .font(.system(size: Constants.sizeFontGetStartedButtonValue))
+                .bold()
+                .foregroundColor(.purple)
+        }
+        .frame(width: Constants.widthGetStartedButtonValue, height: Constants.heightGetStartedButtonValue)
+        .background(getStartedLinearGradientView)
+        .cornerRadius(Constants.cornerRadiusGetStartedValue)
+        .offset(y: Constants.offsetYGetStartedButtonValue)
+    }
+
+    private var havingAccountView: some View {
```

Comment:
```suggestion
    private var havingAccountTextView: some View {
```
ты где-то пишешь Text в названии, а где-то нет